rules/ephemeral/aws_ephemeral_resources.go

```diff
@@ -0,0 +1,90 @@
+package ephemeral
+
+import (
+	"fmt"
+
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+	"github.com/terraform-linters/tflint-ruleset-aws/project"
+)
+
+// AwsEphemeralResourcesRule checks for data sources which can be replaced by ephemeral resources
+type AwsEphemeralResourcesRule struct {
+	tflint.DefaultRule
+
+	replacingEphemeralResources []string
+}
+
+// NewAwsEphemeralResourcesRule returns new rule with default attributes
+func NewAwsEphemeralResourcesRule() *AwsEphemeralResourcesRule {
+	return &AwsEphemeralResourcesRule{
+		replacingEphemeralResources: replacingEphemeralResources,
+	}
+}
+
+// Name returns the rule name
+func (r *AwsEphemeralResourcesRule) Name() string {
+	return "aws_ephemeral_resources"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *AwsEphemeralResourcesRule) Enabled() bool {
+	return false
+}
+
+// Severity returns the rule severity
+func (r *AwsEphemeralResourcesRule) Severity() tflint.Severity {
+	return tflint.WARNING
+}
+
+// Link returns the rule reference link
+func (r *AwsEphemeralResourcesRule) Link() string {
+	return project.ReferenceLink(r.Name())
+}
+
+// Check checks if there is an ephemeral resource which can replace an data source
+func (r *AwsEphemeralResourcesRule) Check(runner tflint.Runner) error {
+	for _, resourceType := range r.replacingEphemeralResources {
+		resources, err := GetDataSourceContent(runner, resourceType, &hclext.BodySchema{}, nil)
+		if err != nil {
+			return err
+		}
+
+		for _, resource := range resources.Blocks {
+			if err := runner.EmitIssueWithFix(
+				r,
+				fmt.Sprintf("\"%s\" is a non-ephemeral data source, which means that all (sensitive) attributes are stored in state. Please use ephemeral resource \"%s\" instead.", resourceType, resourceType),
+				resource.TypeRange,
+				func(f tflint.Fixer) error {
+					return f.ReplaceText(resource.TypeRange, "ephemeral")
```

Comment:
I'm wondering if this autofix is ​​safe. Expressions that reference the data attribute must also be rewritten to `ephemeral.*`, otherwise an error will occur. Furthermore, ephemeral attributes can only be referenced in certain places, in which case only data resources are valid.

Personally, I find this autocorrect too complicated and prefer to start without it.

Reply:
Thanks, I'll remove it.

I didn't know exactly what the qualifications are for auto-fix, but based on this and previous contribution it seems that auto-fix can only be applied if it really fixes all loose ends. That is indeed not the case here, it is only a starting point.